earth_enterprise/src/common/profiler.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 the Open GEE Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PROFILER_H
+#define PROFILER_H
+
+#include <string>
+
+/*
+ * Singleton class for profiling events. The programmer can use this class
+ * to log the timing of operations. This class is intended for performance
+ * debugging.
+ */
+class Profiler {
+  public:
+    static Profiler * instance() { return _instance; }
+    void log(
```

Comment:
can 'log' called from more than one thread ? should be thread safe in general ?

Reply:
Yes, in general this should be thread safe, but that will be implemented in future PRs.